src/selection.rs

```diff
@@ -0,0 +1,112 @@
+use crate::utils::{dist_between_points, Point};
+use std::time::Instant;
+
+type Milliseconds = u128;
+
+const CLICK_TOLERANCE: Milliseconds = 300;
+const MAX_CLICK_DIST: f32 = 5.0;
+
+#[derive(PartialEq, Debug)]
+pub enum SelectionMode {
+    Word,
+    Line,
+}
+
+#[derive(Debug)]
+pub enum SelectionKind {
+    Drag {
+        start: Point,
+        end: Point,
+    },
+    Click {
+        mode: SelectionMode,
+        time: Instant,
+        position: Point,
+    },
+    Start {
+        position: Point,
+        time: Instant,
+    },
+    None,
+}
+
+pub struct Selection {
+    pub selection: SelectionKind,
+    pub text: String,
+}
+impl Selection {
+    pub const fn new() -> Self {
+        Self {
+            selection: SelectionKind::None,
+            text: String::new(),
+        }
+    }
+    pub fn is_none(&self) -> bool {
+        matches!(self.selection, SelectionKind::None)
+    }
+    pub fn start(&mut self, position: Point) {
+        self.selection = SelectionKind::Start {
+            position,
+            time: Instant::now(),
+        }
+    }
+
+    pub fn handle_drag(&mut self, new_position: Point) -> bool {
```

Comment:
```suggestion
    #[must_use]
    pub fn handle_drag(&mut self, new_position: Point) -> bool {
```

Seems easy to forget that this has a meaningful return type